modules/vcffixer.py

```diff
@@ -103,87 +98,79 @@ def process_vcf(vcf_file, dict):
                 alt = splitter[4]
                 # print(ref)
                 # print(alt)
-
-                dupe_check = check_dict_duplicate(dict, pos)
-                if dupe_check == False:
+                if pos in vcf_dict.keys():
+                    pass
+                    #what should happen if position is a duplicate?
+                else:
                     nucs_list.append(ref)
                     nucs_list.append(alt)
                     # print(nucs_list)
-                    dict[int(pos)] = nucs_list
+                    vcf_dict[int(pos)] = nucs_list
+    return vcf_dict
 
-    return dict
 
-
-def compare_and_fix(vcf_dict, align_list, output_list, taxon_name, nuc_set_):
+def compare_and_fix(vcf_dict, cns_list):
     """Reads over alignment sequence and checks if an alternative nucleotide was recorded where mpileup placed an N. \
     The N is replaced by the alternative nucleotide"""
-    print("align list length prior to adjustment: ", len(align_list))
-    # print(align_list)
-    for num, nuc in enumerate(align_list):
+    if _DEBUG:
+        print("align list length prior to adjustment: ", len(cnslist))
+    output_list = []
+    for num, nuc in enumerate(cns_list):
         # print(nuc)
         if nuc.upper() == "N":
             pos_in_vcf = num + 1
-            # ref_and_alt = vcf_dict[pos_in_vcf]
             ref_and_alt = vcf_dict.get(pos_in_vcf)
             if ref_and_alt == None:
-                output_list.append("N")
+                output_list.append(nuc)
             elif ref_and_alt != None:
                 ref = ref_and_alt[0]
-                alt = ref_and_alt[1]
-                split_alts = alt.split(',')
-                if len(split_alts) > 1:
-                    # print(split_alts)
-
-                    # OK, here we just take the first alternative nucleotide
-                    # There is probably a better way to handle this but for now, this works
-                    selected_alt = split_alts[0]
-                    if selected_alt in nuc_set_:
-                        # output_list.append(selected_alt.upper())
-                        align_list[num] = selected_alt.upper()
-                        print("N replaced with {} and position {} from vcf position {}.".format(selected_alt, num, pos_in_vcf))
-
-        # elif nuc.upper() in nuc_set_:
-        #     output_list.append(nuc.upper())
-
-        elif nuc.upper() != "N":
-            if nuc.upper() not in nuc_set_:
-            # output_list.append("N")
-                align_list[num] = "N"
```

Comment:
This is where it was removing degenerate nucleotides - and we can put that back in! But we need to be specific about the choice.